domains/Accounts/Traits/HasRoles.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Domains\Accounts\Traits;
+
+use Domains\Accounts\Enums\AccountTypeEnum;
+
+trait HasRoles
+{
+    public function isOfRole(string $role): bool
+    {
+        return $this->account_type === $role;
+    }
+
+    public function hasRole(string $role): bool
+    {
+        $roles = [];
+        switch ($this->account_type) {
+            case AccountTypeEnum::ADMIN:
+                $roles[] = AccountTypeEnum::ADMIN;
+            case AccountTypeEnum::EDITOR:
+                $roles[] = AccountTypeEnum::EDITOR;
+            case AccountTypeEnum::USER:
+                $roles[] = AccountTypeEnum::USER;
+        }
+
+        return in_array($role, $roles);
+    }
```

Comment:
We only have one role per user, so I guess we only need the `isOfRole` method, because either the `User` instance is related to a given role, or it isn't. Right?

Reply:
The idea here is to have a way to check for permissions without comparing all roles. 

Let's take the case of the change included in this PR to the LinkObserver class:

`optional(Auth::user())->hasRole(AccountTypeEnum::EDITOR)`

I assumed, that an ADMIN should also the same permissions as the EDITOR (+ some more).
Without this hasRole we would have to check 2 roles:

```
optional(Auth::user())->isOfRole(AccountTypeEnum::EDITOR)
|| optional(Auth::user())->isOfRole(AccountTypeEnum::ADMIN)
```

Does it make sense?

Reply:
Ah, I get it. It's a very clever solution. :+1: